packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/SignUpAndSignInDiffSection/utilities.ts

```diff
@@ -0,0 +1,184 @@
+        verificationCode: [{ mutation: 'none', value: oldSignInMethod.verificationCode }],
+      };
+    }
+
+    if (!oldSignInIdentifiers.includes(identifier) && newSignInIdentifiers.includes(identifier)) {
+      const newSignInMethod = newSignInMethods.find(
+        ({ identifier: newIdentifier }) => newIdentifier === identifier
+      );
+
+      assert(newSignInMethod, new Error('Unexpected: the new sign-in method is not found.'));
+
+      return {
+        mutation: 'added',
+        identifier: newSignInMethod.identifier,
+        password: [{ mutation: 'none', value: newSignInMethod.password }],
+        verificationCode: [{ mutation: 'none', value: newSignInMethod.verificationCode }],
+      };
+    }
+
+    const oldSignInMethod = oldSignInMethods.find(
+      ({ identifier: oldIdentifier }) => oldIdentifier === identifier
+    );
+
+    assert(oldSignInMethod, new Error('Unexpected: the old sign-in method is not found.'));
+
+    const newSignInMethod = newSignInMethods.find(
+      ({ identifier: newIdentifier }) => newIdentifier === identifier
+    );
+
+    assert(newSignInMethod, new Error('Unexpected: the new sign-in method is not found.'));
+
+    return {
+      mutation: 'none',
+      identifier: oldSignInMethod.identifier,
+      password:
+        oldSignInMethod.password === newSignInMethod.password
+          ? [{ mutation: 'none', value: oldSignInMethod.password }]
+          : [
+              { mutation: 'removed', value: oldSignInMethod.password },
+              { mutation: 'added', value: newSignInMethod.password },
+            ],
+      verificationCode:
+        oldSignInMethod.verificationCode === newSignInMethod.verificationCode
+          ? [{ mutation: 'none', value: oldSignInMethod.verificationCode }]
+          : [
+              { mutation: 'removed', value: oldSignInMethod.verificationCode },
+              { mutation: 'added', value: newSignInMethod.verificationCode },
+            ],
+    };
+  });
+
+  return signInMethodsDiff
+    .slice()
+    .sort(({ identifier: leftIdentifier }, { identifier: rightIdentifier }) =>
+      stringCompareFunction(leftIdentifier, rightIdentifier)
+    );
+};
+
+export const diffSocialTargets = (
+  oldSocialTargets: string[],
+  newSocialTargets: string[]
+): SocialTargetsDiff => {
+  const allTargets = deduplicate([...oldSocialTargets, ...newSocialTargets]);
+
+  const socialTargetsDiff = allTargets.map<SocialTargetsDiff[number]>((target) => {
+    if (oldSocialTargets.includes(target) && !newSocialTargets.includes(target)) {
+      return { mutation: 'removed', value: target };
+    }
+
+    if (!oldSocialTargets.includes(target) && newSocialTargets.includes(target)) {
+      return { mutation: 'added', value: target };
+    }
+
+    return { mutation: 'none', value: target };
+  });
+
+  return socialTargetsDiff
+    .slice()
+    .sort(({ value: leftValue }, { value: rightValue }) =>
+      stringCompareFunction(leftValue, rightValue)
+    );
```

Comment:
```suggestion
  return socialTargetsDiff
    .slice()
    .sort(({ value: leftValue }, { value: rightValue }) =>
      stringCompareFunction(leftValue, rightValue)
    );
```
Can be simplified to:
```ts
return socialTargetsDiff.slice().sort(stringCompareFunction);
```

Reply:
No, the params of the sort function are objects, not just simple strings.